src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+            # TODO remove the commented block if the projected points loading is not required
+            # flatten the list of lists
+            targets = list(
+                itertools.chain.from_iterable([x] if not isinstance(x, list) else x for x in targets))
+            self._tab_data_model.main.targets.value = targets
+            if self.correlation_target.fib_surface_fiducial:
+                self._tab_data_model.fib_surface_point.value = self.correlation_target.fib_surface_fiducial
+        else:
+            # initialize the correlation target
+            self._tab_data_model.main.currentFeature.value.correlation_targets = {}
+            self._tab_data_model.main.currentFeature.value.correlation_targets[
+                self._tab_data_model.main.currentFeature.value.status.value] = CorrelationTarget()
+            self.correlation_target = self._tab_data_model.main.currentFeature.value.correlation_targets[
+                self._tab_data_model.main.currentFeature.value.status.value]
+            self.group_add_streams()
+
+        self._panel.fp_correlation_panel.Show(True)
+        for vp in self._viewports:
+            vp.canvas.Bind(wx.EVT_CHAR, self._on_char)
+
+    @call_in_wx_main
+    def _on_fm_streams_change(self, stream_projections: ListVA) -> None:
+        """
+        Update the stream groups when the user selects the FM stream panel from a different group. The
+        group of the selected stream panel will be visible and the other groups will be invisible
+        as well as collapsed.
+        """
+        if not self.stream_groups or len(stream_projections) <= 0:
+            return
+
+        stream_obj = stream_projections[-1].stream  # Get the stream name
+
+        # Find the index of the stream in the stream list
+        stream_index = next(
+            (i for i, stream in enumerate(self._tab_data_model.main.currentFeature.value.streams.value)
+             if stream == stream_obj), None)
+
+        # If the stream exists in the list
+        if stream_index is not None:
+            # Get the group this stream belongs to
+            group_key = next(
+                (key for key, indices in self.stream_groups.items() if stream_index in indices), None)
+
+            # Update visibility for the group
+            if self.previous_group == group_key:
+                return
+
+            self.previous_group = group_key
+            if group_key:
+                streams_list = self._tab_data_model.main.currentFeature.value.streams.value
+                for key, indices in self.stream_groups.items():
+                    for index in indices:
+                        stream = streams_list[index]
+                        # find stream controller for the stream
+                        ssc = next(
+                            (sc for sc in  self._panel.streambar_controller.stream_controllers
+                             if sc.stream == stream), None)
```

Comment:
ssc will be `None` if no controller can be found. Probably very unlikely, but you should still explicitly handle such case, maybe just with a error log and ending the function.